src/java.desktop/share/classes/javax/swing/plaf/synth/SynthSplitPaneUI.java

```diff
@@ -265,6 +269,16 @@ protected Component createDefaultNonContinuousLayoutDivider() {
         return new Canvas() {
             public void paint(Graphics g) {
                 paintDragDivider(g, 0, 0, getWidth(), getHeight());
+                if(!isContinuousLayout() && getLastDragLocation() != -1) {
+                    Dimension      size = splitPane.getSize();
+
+                    g.setColor(dividerDraggingColor);
+                    if(getOrientation() == JSplitPane.HORIZONTAL_SPLIT) {
+                        g.fillRect(0, 0, dividerSize - 1, size.height - 1);
+                    } else {
+                        g.fillRect(0, 0, size.width - 1, dividerSize - 1);
+                    }
+                }
```

Comment:
This piece of code is basically same as the code in `finishedPaintingChildren`. Can the duplication be eliminated by introducing a helper method?

Reply:
helper method added